### ISSUE TITLE
Accept an array of responses in `mockClient`

### ISSUE THREAD

**Issue by tulios** [tulios/mappersmith#39]
It can be really useful if `mockClient` accepts an array of responses to return in order

```javascript
mockClient(client)
  .resource('User')
  .method('all')
  .response([
    { allUsers: [{id: 1}] },
    { allUsers: [{id: 1, 2}] },
    { allUsers: [{id: 2, 3, 4}] }
  ])
```

It will return specific values in order until it reaches the end of the response values list, at which point it will throw an error for all subsequent calls.

**Comment by tulios** [comment]
https://github.com/tulios/mappersmith/tree/add-array-of-responses-to-mock-client

**Comment by jbeuckm** [comment]
I did this as a callback. If we just sent an array, the mock would not know whether to respond with that array or sequentially with elements from that array.

**Comment by tulios** [comment]
Fixed in #91
